sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c

```diff
@@ -26,6 +26,36 @@
 #include "debug.h"
 
 #ifdef MODULE_GNRC_IPV6_EXT_RH
+/* unchecked precondition: hdr is gnrc_pktsnip_t::data of the
+ * GNRC_NETTYPE_IPV6 snip within pkt */
+static void _forward_pkt(gnrc_pktsnip_t *pkt, ipv6_hdr_t *hdr)
+{
+    gnrc_pktsnip_t *netif_snip;
+
+    if (--(hdr->hl) == 0) {
+        DEBUG("ipv6_ext_rh: hop limit reached 0: drop packet\n");
+        gnrc_pktbuf_release(pkt);
+    }
+    /* remove L2 headers around IPV6 */
+    netif_snip = gnrc_pktsnip_search_type(pkt, GNRC_NETTYPE_NETIF);
+    if (netif_snip != NULL) {
+        gnrc_pktbuf_remove_snip(pkt, netif_snip);
```

Comment:
Shouldn't this be something like `pkt = gnrc_pktbuf_remove_snip(pkt, netif_snip);`?

Reply:
Technically you are right, but since it is highly unlikely, that the `netif_snip` is at the start of the list at this point, it isn't necessary. For consistency sake I will however do it.